types/schemas/discriminated/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ApplicationTypeKeys,
+  PrimaryApplicationType,
+} from '../digitalPlanningApplication/enums/ApplicationTypes';
+import {ApplicationData} from './ApplicationData';
+import {User} from './User';
+
+/**
+ * @internal
+ * The generic base type for all applications
+ * Takes a primary and granular application type which allows child properties to differ based on these inputs
+ * Deriving `TPrimary` from `TGranular` is possible in TS, but not in a way which is currently compatible with ts-json-schema-generator
+ */
+interface ApplicationSpecification<
+  TPrimary extends PrimaryApplicationType,
+  TGranular extends ApplicationTypeKeys,
+> {
+  applicationType: TGranular;
```

Comment:
This is required to be above or at an equal hierarchy level of anywhere `TGranular` or `TPrimary` is passed in.

This means if we need variants based on `TGranular` or `TPrimary` outside of `data.application` (I think we do?), we can't just keep this in `data.application.applicationType`.